ross/rotor_assembly.py

```diff
@@ -483,11 +487,14 @@ def flatten(l):
                 y_pos_sup += 2 * mean_od * df["scale_factor"][df.tag == t].values[0]
 
         # define position for point mass elements
+        # fmt: off
         dfb = df[
-            (df.type == "BearingElement")
-            | (df.type == "BearingElement6DoF")
-            | (df.type == "SealElement")
+            df.type.isin([
+                "BallBearingElement", "BearingElement", "BearingFluidFlow",
+                "BearingElement6DoF", "MagneticBearingElement", "RollerBearingElement",
+                "SealElement"])
```

Comment:
To avoid having to edit this in the future, maybe we could do something like:
```python
dfb = df[df.apply(lambda x: isinstance(x, BearingElement))]
```

Reply:
It doesn't work.
`df.type` is column of `str` types, so it always returns `False`

Reply:
I found another way, but we need to import `sys` and `inspect`.

```python
classes = [_class for _class, _ in inspect.getmembers(sys.modules["ross.bearing_seal_element"], inspect.isclass)]
dfb = df[df.type.isin(classes)]
```

Reply:
This is good for me!

Reply:
i can replace some unused imports on `rotor_assembly.py` like `os` and `shutil`